### DISCUSSION TITLE
problem parsing async function

### DISCUSSION THREAD

**Discussion by mgotrik** [SilentVoid13/Templater#990] [Help]
Hello,
 I am trying to write a function that takes in a date and starts searching backwards until it finds a note with that date as a header. i.e. provide it 2022-12-31 and it would start looking for notes with 2022-12-30, 2022-12-29... until it finds a note that exists. 

I thought things were going smoothly until I realized that tp.file.exists() yields a Promise and has to be resolved asynchronously. When I try to convert my function to an async function i keep getting "Template parsing error". I am a total newb at JS so maybe this is a dumb question.

Anyways, here is my code:

```
async filenameOnOrBefore(date, filepath, tp) {
    var input = new Date(date);
    var dateString = input.toISOString().slice(0, 10);
    var filename = filepath + dateString + ".md";
    var count = 0;
    while (!(await tp.file.exists(filename)) && count < 30) {
        input.setDate(input.getDate() - 1);
        dateString = input.toISOString().slice(0, 10);
        filename = filepath + dateString + ".md";
        count++;
    }
    if(count === 30) return filename;
    return filename;
}
module.exports = filenameOnOrBefore;
```
Appreciate any help I can get.

**Comment by AB1908** [comment]
Could you paste the error message as well?

**Comment by mgotrik** [comment]
The error just says "Templater parsing error, aborting. Check console for more information."

```
app.js:1 Obsidian Developer Console
plugin:obsidian-tasks-plugin:163 loading plugin "tasks"
plugin:obsidian-day-planner:7214 Loading Day Planner plugin
plugin:templater-obsidian:61 Templater Error: Template parsing error, aborting. 
 Unexpected identifier
log_error @ plugin:templater-obsidian:61
errorWrapper @ plugin:templater-obsidian:81
await in errorWrapper (async)
append_template_to_active_file @ plugin:templater-obsidian:3700
callback @ plugin:templater-obsidian:3978
XN @ app.js:1
e.executeCommand @ app.js:1
e.onTrigger @ app.js:1
e.handleKey @ app.js:1
e.handleKey @ app.js:1
t.handleKey @ app.js:1
e.onKeyEvent @ app.js:1
```
Edit: I figured it out - I just changed the line from `async filenameOnOrBefore(date, filepath, tp) {` to `async function filenameOnOrBefore(date, filepath, tp) {`
